guides/puppet-web-apps/html-template-guidelines/README.md

```diff
@@ -0,0 +1,157 @@
+
+Slot names are added as attributes to the functional elements like so:
+
+```html
+<button slot="MyApp/SubmitButton" value="{{model.Submit::click}}">Submit</button>
+
+<div slot="MyApp/LifeReport">
+    <template is="dom-if" if="{{model.Person.Alive}}">
+        <strong>Still going strong</strong>
+    </template>
+</div>
+```
+
+### Guideline 4: Create Layout in `starcounter-composition`
+
+As outlined in guideline 1, the layout of the HTML template should be included within the `<template is="starcounter-composition">`.
+
+The elements with a slot are grabbed into the `starcounter-composition` using the following syntax: `<content select="[slot='AppName/ElementName']"></content>`.
+
+Consider the following template only containing functional elements:
+
+```html
+<link rel="import" href="/sys/polymer/polymer.html">
+
+<template>
+    <template is="dom-bind">
+        <h1 slot="MyApp/ImportantHeadline">Very Important Headline</h1>
+        <div slot="MyApp/LifeReport">
+            <template is="dom-if" if="{{model.Person.Alive}}">
+                <strong>Still going strong</strong>
+            </template>
+        </div>
+        <input slot="MyApp/StatusInput" value="{{model.Status$::input}}" />
+    </template>
+</template>
+```
+
+To add a `starcounter-composition` to this template, something like this can be done:
+
+```html
+<link rel="import" href="/sys/polymer/polymer.html">
+
+<template>
+    <template is="dom-bind">
+        <h1 slot="MyApp/ImportantHeadline">Very Important Headline</h1>
+        <div slot="MyApp/LifeReport">
+            <template is="dom-if" if="{{model.Person.Alive}}">
+                <strong>Still going strong</strong>
+            </template>
+        </div>
+        <input slot="MyApp/StatusInput" value="{{model.Status$::input}}" />
+    </template>
+    <template is="starcounter-composition">
+        <content select="[slot='MyApp/ImportantHeadline']"></content>
+        <div class="myapp-life-report">
+            <content select="[slot='MyApp/LifeReport']"></content>
+            <content select="[slot='MyApp/StatusInput']"></content>
+        </div>
+    </template>
+</template>
+```
+
+### Guideline 5: Apply Styling to Avoid Conflicts and Allow Blending
+
+Regarding styling, there are two ways to make the application easier to visually integrate with other apps:
+
+1. Prefix the class with the name of the app. As outlined in [Avoiding CSS Conflicts](https://docs.starcounter.io/guides/mapping-and-blending/avoiding-css-conflicts/), the class should be prefixed with the name of the app to avoid CSS conflicts with classes from other apps.
+
+2. Keep styling that will affect the layout inside the `starcounter-composition`. This includes CSS attributes such as `padding`, `margin`, `display` and `float`.

```

Comment:
Class names used in `starcounter-composition` should also be prefixed.

Also all CSS attributes applies to `sc-composition` so I see no point in listing those few.